src/Microsoft.OData.Core/JsonLight/ODataJsonLightResourceDeserializer.cs

```diff
@@ -1552,7 +3685,12 @@ private ODataJsonLightReaderNestedInfo InnerReadUndeclaredProperty(IODataJsonLig
             if (payloadTypeOrItemType != null)
             {
                 // Complex property or collection of complex property.
-                ValidateExpandedNestedResourceInfoPropertyValue(this.JsonReader, isCollection, propertyName, payloadTypeReference);
+                await ValidateExpandedNestedResourceInfoPropertyValueAsync(
+                    this.JsonReader.NodeType,
+                    () => this.JsonReader.GetValueAsync(),

```

Comment:
Similar questions as before regarding the use of a delegate here instead of just passing the json reader directly.

Reply:
Fixed by removing the delegate